### DISCUSSION TITLE
Menu Action closes when pressing Enter.

### DISCUSSION THREAD

**Discussion by VilikNakhshikyan** [eclipse-theia/theia#14071] [General] [answered]
I have my custom Theia application and I added a new menu item to it using registerMenuAction.

Theia closes my menu when I press Enter, Is there a way to unregister this Keybinding.

**Comment by msujew** [comment]
Hey @VilikNakhshikyan,

I believe when pressing `Enter`, you're not exactly triggering a keybinding (we don't have a general one that triggers on `Enter`), but either something in [Phosphor.js](https://github.com/phosphorjs/phosphor) (soon to be replaced with Lumino) or something in the DOM API. Both of which we don't have any control over.

**Reply by VilikNakhshikyan** [reply]
I have registered theia dialog to my menu item and pressing enter closes the theia dialog even if the mouse is not on the 'close' button of theia dialog.  

Do you have any ideas?

**Reply by msujew** [reply, marked as answer]
Oh, ok. Please give a bit more context next time. We have some special handling for key events on dialogs. See:

https://github.com/eclipse-theia/theia/blob/acaa3dfb6733665af5ab6ab9025fdf37ea758668/packages/core/src/browser/dialogs.ts#L247-L252

You will likely need to overwrite this method to change the behavior in your dialog.

**Reply by VilikNakhshikyan** [reply]
Thank you very much the problem was solved. Next time I will be more detailed. :)
